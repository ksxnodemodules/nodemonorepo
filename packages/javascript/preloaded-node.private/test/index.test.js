'use strict'
const path = require('path')
const xjest = require('extra-jest')
const {spawn, spawnSync} = require('../index')
const script = path.resolve(__dirname, 'data/main.js')

beforeEach(() => {
  jest.setTimeout(131072)
})

afterEach(() => {
  jest.setTimeout(5000)
})

it('spawn function works', () => {
  let stdout = ''
  let stderr = ''
  const child = spawn([script], {encoding: 'utf8'})

  child.stdout.on('data', chunk => { stdout += chunk })
  child.stderr.on('data', chunk => { stderr += chunk })

  return new Promise((resolve, reject) => {
    child.on('exit', status => resolve(status))
    child.on('error', error => reject(error))
  }).then(
    status => xjest.snap.unsafe({stdout, stderr, status})()
  )
})

it('spawnSync function works', () => {
  const {status, error, signal, stdout, stderr} = spawnSync([script], {encoding: 'utf8'})
  xjest.snap.unsafe({status, error, signal, stdout, stderr})()
})
