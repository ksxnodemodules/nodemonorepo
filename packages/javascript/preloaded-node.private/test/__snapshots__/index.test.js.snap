// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spawn function works 1`] = `
"
status: 123
stderr: ''
stdout: |
  { TypeScript: 'Hello World !!' }

"
`;

exports[`spawnSync function works 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 123
stderr: ''
stdout: |
  { TypeScript: 'Hello World !!' }

"
`;
